internal/core/services/operation_manager/operation_manager.go

```diff
@@ -267,13 +267,22 @@ func (om *OperationManager) StartLeaseRenewGoRoutine(operationCtx context.Contex
 }
 
 func (om *OperationManager) addOperationsLeaseData(ctx context.Context, schedulerName string, ops []*operation.Operation) error {
+	opMap := make(map[string]*operation.Operation)
+	opIds := make([]string, 0, len(ops))
 	for _, op := range ops {
-		ttl, err := om.leaseStorage.FetchLeaseTTL(ctx, schedulerName, op.ID)
-		if err != nil {
-			return fmt.Errorf("failed to fetch lease data for scheduler %s operation %s: %w", op.ID, schedulerName, err)
-		}
-		op.Lease = operation.OperationLease{OperationID: op.ID, Ttl: ttl}
+		opMap[op.ID] = op
+		opIds = append(opIds, op.ID)
+	}
+
+	leases, err := om.leaseStorage.FetchOperationsLease(ctx, schedulerName, opIds...)
```

Comment:
Although it might be unusual (a rare case) but wouldn't be prudent to check if len(opIds) > 0 before calling this method? Actually seems a validation we could do even before calling **addOperationsLeaseData**...

Reply:
Definitly!